src/aiko_services/main/scheme.py

```diff
@@ -48,12 +48,16 @@ def destroy_targets(self, stream):
 
     @classmethod
     def parse_data_url_path(cls, data_url):  # data_source or data_target
+        if data_url.startswith("data:"):
+            raise NotImplementedError("'data:' URLs do not have a path component")
```

Comment:
The URL specification effectively states that all URLs have a `path` component ... including the `data:` scheme.
Therefore,  `parse_data_url_path()` should always return a valid `path` and not raise an Exception.
Note: For the `data:` scheme it is invalid to have an empty `path` and there must be at least a comma that separates the (optional) media type / parameters from the data, e.g `data:,`

Time for me to incorporate the standard Python URL parsing library function call.